### PR TITLE
feat(ui) Build entity doesn't exist page for entity profiles

### PR DESCRIPTION
**Pull request by chriscollins3456**
In the embedded profile for the chrome extension as well as any entity profile, we want to show a page saying that an entity doesn't exist in datahub if that's the case instead of pretending like we know about the entity and showing a bare bones page (like we do in the main app).

This adds a new `exists` field on entities that you can query in graphql. It reuses an existing `EntityExistsResolver` that was meant as its own endpoint with a `urn` in the params, but I repurpose it to get the urn from the source entity if it's not in the params so it can be used for both purposes.

Here's what the page looks like for embedded profiles:
image

and for regular profiles:
image

## Checklist

- [ ] The PR conforms to DataHub's [Contributing Guideline](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [ ] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable). If a new feature has been added a Usage Guide has been added for the same.
- [ ] For any breaking change/potential downtime/deprecation/big changes an entry has been made in [Updating DataHub](https://github.com/datahub-project/datahub/blob/master/docs/how/updating-datahub.md)

### CODE REVIEW

File: datahub-graphql-core/src/main/java/com/linkedin/datahub/graphql/GmsGraphQLEngine.java [hunk 1]
```diff
@@ -1256,7 +1263,9 @@ private void configureTypeResolvers(final RuntimeWiring.Builder builder) {
                         .filter(graphType -> graphType instanceof EntityType)
                         .map(graphType -> (EntityType<?, ?>) graphType)
                         .collect(Collectors.toList())
-                )))
+                ))
+                .dataFetcher("exists", new EntityExistsResolver(entityService))
```
Comment (jjoyce0510) [inline datahub-graphql-core/src/main/java/com/linkedin/datahub/graphql/GmsGraphQLEngine.java:1267]: I'm not sure we need to do this here. I think it should suffice to have resolvers registered on each individual entity type.

File: datahub-graphql-core/src/main/java/com/linkedin/datahub/graphql/types/dataset/mappers/DatasetMapper.java [hunk 2]
```diff
@@ -73,6 +73,13 @@ public Dataset apply(@Nonnull final EntityResponse entityResponse) {
         Long lastIngested = SystemMetadataUtils.getLastIngested(aspectMap);
         result.setLastIngested(lastIngested);
 
+        System.out.println("~~~~~~~~~~~~~~~~~~~~~~HERE~~~~~~~~~~~~~~~");
```
Comment (jjoyce0510) [inline datahub-graphql-core/src/main/java/com/linkedin/datahub/graphql/types/dataset/mappers/DatasetMapper.java:76]: Let's remove this

File: datahub-graphql-core/src/main/java/com/linkedin/datahub/graphql/types/dataset/mappers/DatasetMapper.java [hunk 3]
```diff
@@ -110,6 +117,12 @@ public Dataset apply(@Nonnull final EntityResponse entityResponse) {
 
     private void mapDatasetKey(@Nonnull Dataset dataset, @Nonnull DataMap dataMap) {
         final DatasetKey gmsKey = new DatasetKey(dataMap);
+        System.out.println("~~~~~~~~~~~~~~~~~~~~~~NOW~~~~~~~~~~~~~~~");
+        System.out.println(gmsKey);
```
Comment (jjoyce0510) [inline datahub-graphql-core/src/main/java/com/linkedin/datahub/graphql/types/dataset/mappers/DatasetMapper.java:121]: sames

File: datahub-graphql-core/src/main/resources/entity.graphql [hunk 4]
```diff
@@ -1241,6 +1241,11 @@ type Dataset implements EntityWithRelationships & Entity & BrowsableEntity {
     Privileges given to a user relevant to this entity
     """
     privileges: EntityPrivileges
+
+    """
+    Whether or not this entity exists in the database
```
Comment (jjoyce0510) [inline datahub-graphql-core/src/main/resources/entity.graphql:1246]: in the database -> on DataHub

File: datahub-web-react/src/app/entity/shared/embed/EmbeddedProfile.tsx [hunk 5]
```diff
@@ -49,6 +49,10 @@ export default function EmbeddedProfile<T>({ urn, entityType, getOverridePropert
     const { entityData, dataPossiblyCombinedWithSiblings, dataNotCombinedWithSiblings, loading, refetch } =
         useGetDataForProfile({ urn, entityType, useEntityQuery, getOverrideProperties });
 
+    if (entityData?.exists === false) {
```
Comment (jjoyce0510) [inline datahub-web-react/src/app/entity/shared/embed/EmbeddedProfile.tsx:52]: awesome!

File: datahub-web-react/src/graphql/chart.graphql [hunk 6]
```diff
@@ -2,6 +2,7 @@ query getChart($urn: String!) {
     chart(urn: $urn) {
         urn
         type
+        exists
```
Comment (jjoyce0510) [inline datahub-web-react/src/graphql/chart.graphql:5]: do we need to add this to dashboard.graphql as well?

File: datahub-web-react/src/graphql/dataFlow.graphql [hunk 7]
```diff
@@ -1,6 +1,7 @@
 fragment dataFlowFields on DataFlow {
     urn
     type
+    exists
```
Comment (jjoyce0510) [inline datahub-web-react/src/graphql/dataFlow.graphql:4]: how about dataJob fields?

File: datahub-web-react/src/graphql/chart.graphql (hunk 6 above)
Comment (jjoyce0510) [reply datahub-web-react/src/graphql/chart.graphql:5]: (i see its in fragments)

**Review by jjoyce0510 — APPROVED** [review]
Were you planning on adding this inside of EntityProfile component as part of this PR or separately? 

Looks like FE build failing!

**Comment by chriscollins3456** [comment]
> Were you planning on adding this inside of EntityProfile component as part of this PR or separately?

Shoot I just completely forgot to push up my last commit adding this to the regular profie!

File: datahub-graphql-core/src/main/java/com/linkedin/datahub/graphql/types/dataset/mappers/DatasetMapper.java (hunk 3 above)
Comment (chriscollins3456) [reply datahub-graphql-core/src/main/java/com/linkedin/datahub/graphql/types/dataset/mappers/DatasetMapper.java:121]: this is what i get for rushing a PR up at the end of the day haha

File: datahub-web-react/src/graphql/chart.graphql (hunk 6 above)
Comment (chriscollins3456) [reply datahub-web-react/src/graphql/chart.graphql:5]: yup I got it in `dashboardFields` in fragments.graphql!

**Review by jjoyce0510 — APPROVED** [review]
Really really nice, man!

**Pull request merged**